Kendall Brown
8564403
Pstat 134 
Spring 2018
Assignment 1

For this assignment we will be verifying that a sample drawn from N($\mu$,$\sigma^2$) can be standardized to represent a sample from a distribution of N(0,1). In order to accomplish this we will draw a random sample of 1,00,000 values from the arbitrarily chosen N(100,10) distribution and compute the sample mean and variance.  Simulated values will be rounded to 3 decimal places.

In [1]:
import numpy as np
import pandas as pd
import scipy as sp
# Firstly, we set a seed of 100 for ensure repeatability
np.random.seed(300)
# Secondly, we construct an array named "x" to hold the simulated outcomes of the random variable X.  
x=np.random.normal(100,10,1000000)
x_round=np.round(x,3)
print(x_round)
# Calculating sample mean
x_mean=np.mean(x_round)
# Calculating sample variance
x_var=np.var(x_round)
# Sample mean
print(x_mean)
# Sample Variance
print(x_var)
# Bottom 2.5% quantile
x_lb=np.percentile(x_round,2.5)
print(x_lb)
# Top 2.5% quantile
x_ub=np.percentile(x_round,97.5)
print(x_ub)

[  85.148   97.398   85.754 ...,   98.229  103.747   73.686]
100.000325885
99.9680378791
80.403975
119.577


From our simulation we observe a sample mean of 100.000 and sample variance of 99.97. Compared to the theoretical mean and variance 100 and 100 respecively, we take the sample mean to be an accurate estimate of the theoretical mean. Additionally, our simulated data shows that we have a 95% of values between 80.403975 and 119.577.    

To verify if the sample distribution can be standardized to be N(0,1) we will preform a transformation on each data point. 
Firstly, we will subtract the sample mean from each data point. Then, the difference will be divided my the sample standard deviation.

In [2]:
x_stand=x_round
for i in range(len(x_round)):
    x_stand[i]=((x_round[i]-x_mean)/((x_var)**.5))
print(x_stand)
# Calculating sample mean of standardized data
x_mean_std=np.mean(x_stand)
# Calculating sample variance of standardized data
x_var_std=np.var(x_stand)
# Sample mean
print(x_mean_std)
# Sample Variance
print(x_var_std)
# Bottom 2.5% quantile
x_std_lb=np.percentile(x_stand,2.5)
print(x_std_lb)
# Top 2.5% quantile
x_std_ub=np.percentile(x_stand,97.5)
print(x_std_ub)

[-1.48547    -0.26027419 -1.42486031 ..., -0.1771609   0.3747273
 -2.63185322]
-3.78290110348e-15
1.0
-1.95994833406
1.95798034253


From the transformed data we see that the new mean is approximately 0 and the standard deviation is 1. Additionally, we see 95% of observation lay between -1.95995 and 1.95798. Comparing to the theoretical 95% confidence interval of the N(0,1) distribution, {-1.96,1.96}, we are rather confident that we have represented the N(100,10) distribution as a N(0,1) distribution.  

As a measure of extra confidence we will preform the k-s test to compare our standardized data to the N(0,1) distribution. This test will measure the validity of the hypothesis that the standardized sample follows distribution N(0,1). We  will set the decision boundry for our test to be $\alpha$=.05. 

In [3]:
from scipy import stats
stats.kstest(x_stand,'norm') # here 'norm' represents distribution N(0,1)

KstestResult(statistic=0.00076251584645464776, pvalue=0.6061426149324487)

We observed a test statistic of ~.0008 anda p-value of .60614. This observed p-value exceeds our $\alpha$ level quite significantly. We fail to reject the null hypothesis that the standardized data set follows N(0,1).
Thus we conclude that it is possible to standardize a data set of N($\mu$,$\sigma^2$) to follow distribution N(0,1).